WordPress/gutenberg
# Conditional Duotone Toolbar Visibilty

## Description


A block supporting duotone needs to be able to control the visibility of the toolbar. Data needs to be passed from the supporting block (i.e. image block) to the block supports (i.e. duotone).

See #31373 

> First off, the feature should not be visible in the setup state

> Secondly, the Duotone button should always be at the end. Right now it starts as the first button, then if you deselect the block and select it again, it moves to the end

This change allows the image block to prevent rendering of the duotone toolbar. This has the side-effect of solving the second issue since the point where the duotone toolbar is rendered comes after the rest of the block toolbar items, so the toolbar button is now always rendered at the end.

## How has this been tested?




1. Create a cover block
2. See that there isn't a duotone button
3. Add a background image
4. See that there is a duotone button
5. Enable parallax/repeat backgrounds or clear media 
6. See that duotone button is hidden again
7. Repeat 1-4 for image block instead

## Screenshots 

<p align="center"><img width="629" alt="image" src="https://user-images.githubusercontent.com/5129775/117212271-d6c68980-ad95-11eb-92a2-feb7de4763c2.png"></p>

<p align="center"><img width="646" alt="image" src="https://user-images.githubusercontent.com/5129775/117212371-f78edf00-ad95-11eb-9ae9-732c8292158e.png"></p>

## Types of changes





New feature / Bug fix

## Checklist:
